### PR TITLE
(#249) CMS Card Title Verification tests

### LINKED ISSUE #249: CMS: Card Title verification (in Mini Landing page)
## As a CMS user, I want to verify that card title used in different content types (Press Releases, Events, Biography etc.) appear correctly on Mini Landing pages and Home/Landing pages. 

In order to verify Card title field functionality, every created content type should be added to the 
a) Mini Landing page using Add Content Block -> Add Two Item Feature Card Row-> Internal feature Card.
Every separate content feature will include a scenario to create a basic mini landing, add a test url of the newly created content into Internal Feature Card, publish and verify that the card title provided in the newly created content is displayed in the mini landing page.

> **ESTIMATE** TBD

### Acceptance Criteria

NOTE!!!! This should be worked on after https://github.com/NCIOCPL/cgov-digital-platform-acceptance-tests/issues/248 is completed. This ensures that all the necessary background work has already been added to the features

ADD BELOW STEP TO     Scenario: Verify promo and card titles in mini landing. IN EVERY CONTENT TYPE THAT HAS IT 

```gherkin 
        Then the card title is "replace this with a card title value  you entered earlier"
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Closes #249 
Added tests for Article

### CODE REVIEW

**Comment by alinai** [comment]
Cypress test run on dev-ac:
image
image
image
image
image

**Comment by alinai** [comment]
Cypress test run on dev-ac:
image
image
image
image

**Comment by alinai** [comment]
Cypress test run on dev-ac:
image

**Comment by alinai** [comment]
cypress test run on dev-ac:
image

File: cypress/e2e/common/cmsSteps.js [hunk 1]
```diff
@@ -494,4 +494,16 @@ And('preview button was translated as {string}', (previewBtn) => {
...
 And('date published was translated as {string}', (datePublished) => {
     cy.get('ul.clearfix li strong').should('be.visible').and('include.text', datePublished);
+});
+
+And('the Card Title has a link {string} with href {string}', (linkText, linkHref) => {
+    cy.get(`a[href="${linkHref}"] h3:contains('${linkText}')`).should('be.visible');
+});
+
+And('feature card description reads {string}', (description) => {
+    cy.get(`p:contains('${description}')`).should('be.visible');
```
Comment (KateMashkinaNIH) [inline cypress/e2e/common/cmsSteps.js:504]: could you make this locator more specific to point to feature card 
like .feature-card p:contains('${description}')

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

File: cypress/e2e/common/cmsSteps.js [hunk 2]
```diff
@@ -494,4 +494,16 @@ And('preview button was translated as {string}', (previewBtn) => {
 
 And('date published was translated as {string}', (datePublished) => {
     cy.get('ul.clearfix li strong').should('be.visible').and('include.text', datePublished);
+});
+
+And('the Card Title has a link {string} with href {string}', (linkText, linkHref) => {
+    cy.get(`a[href="${linkHref}"] h3:contains('${linkText}')`).should('be.visible');
```
[the remark this answers is not in the document]
Comment (alinai) [reply cypress/e2e/common/cmsSteps.js:500]: done

File: cypress/e2e/common/cmsSteps.js (hunk 1 above)
Comment (alinai) [reply cypress/e2e/common/cmsSteps.js:504]: done

**Comment by alinai** [comment]
npn test run after updating per review:

image

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**